lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereConnection.php

```diff
@@ -137,13 +130,29 @@ public function getServerVersion()
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
         if ($name === null) {
-            return sasql_insert_id($this->connection);
+            $result = sasql_insert_id($this->connection);
+
+            if ($result === false) {
+                throw new SQLAnywhereException('The connection is not valid.');
```

Comment:
The connection is established in the constructor. If there's a run-time error, it shouldn't be hidden but instead reflected in the exception.

Reply:
From the [docs](http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc01776.1600/doc/html/san1357754986165.html), it looks like this method can only return an error (`false`) *"if the $conn is not valid*". So IMO the error message should just reflect that (for example if the connection was closed in the meantime).

Reply:
An invalid connection may have many reasons behind it. If it was valid at the construction time but isn't anymore, I want to know the exact reason (should be available via `SQLAnywhereException::fromSQLAnywhereError($this->connection)`).

Reply:
I didn't know about this method, fixed!